core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1183,6 +1183,21 @@ Builder removeSpecs(Iterable<Integer> specIds) {
               .filter(s -> !specIdsToRemove.contains(s.specId()))
               .collect(Collectors.toList());
       changes.add(new MetadataUpdate.RemovePartitionSpecs(specIdsToRemove));
+
+      return this;
+    }
+
+    Builder removeSchemas(Iterable<Integer> schemaIds) {
+      Set<Integer> schemaIdsToRemove = Sets.newHashSet(schemaIds);
+      Preconditions.checkArgument(
+          !schemaIdsToRemove.contains(currentSchemaId), "Cannot remove the current schema");
+
+      this.schemas =
+          schemas.stream()
+              .filter(s -> !schemaIdsToRemove.contains(s.schemaId()))
+              .collect(Collectors.toList());
+      changes.add(new MetadataUpdate.RemoveSchemas(schemaIdsToRemove));
```

Comment:
I think we should only create a change when `schemaIdsToRemove` was non-empty. This would most likely also apply for removePartitionSpecs

Reply:
Good point! Done

Will create a separate PR for the same for RemovePartitionSpecs once we sort this one out.

Reply:
I prepared another PR for the same RemovePartitionSpecs improvement. Will publish it once this is merged (that also depends on the TestTables change in his PR and not sure how to publish stacked changes as separate PRs TBH)